docker/daemon/daemon_src.go

```diff
@@ -349,7 +362,28 @@ func (s *daemonImageSource) GetBlob(info types.BlobInfo) (io.ReadCloser, int64,
 		if err != nil {
 			return nil, 0, err
 		}
-		return stream, li.size, nil
+
+		// In order to handle the fact that digests != diffIDs (and thus that a
+		// caller which is trying to verify the blob will run into problems),
+		// we need to decompress blobs. This is a bit ugly, but it's a
+		// consequence of making everything addressable by their DiffID rather
+		// than by their digest...
+		decompressFunc, reader, err := compression.DetectCompression(stream)
+		if err != nil {
+			return nil, 0, errors.Wrapf(err, "detecting compression in blob %s", info.Digest)
+		}
```

Comment:
No, because "no compression" is considered a valid "compressor" if you look in `pkg/compression`. You only get errors if you get an EOF that was unexpected for example. However, I need to add a check that `decompressorFunc != nil`.